ci/build-test-matrix.js

```diff
@@ -5,154 +5,241 @@
+        "wasmtime-wasi-nn": ["onnx"]
+      },
+    },
+    {
+      "os": "ubuntu-latest",
+      "name": "Test MSRV on Linux x86_64",
+      "filter": "linux-x64",
+      "isa": "x64",
+      "rust": "msrv",
+    },
+    {
+      "os": "ubuntu-latest",
+      "name": "Test Linux x86_64 with MPK",
+      "filter": "linux-x64",
+      "isa": "x64"
+    },
+    {
+      "os": "macos-13",
+      "name": "Test macOS x86_64",
+      "filter": "macos-x64",
+      "extra_features": {
+        "wasmtime-wasi-nn": ["onnx"]
+      },
+    },
+    {
+      "os": "macos-14",
+      "name": "Test macOS arm64",
+      "filter": "macos-arm64",
+      "target": "aarch64-apple-darwin",
+      "extra_features": {
+        "wasmtime-wasi-nn": ["onnx"]
+      },
+    },
+    {
+      "os": "windows-latest",
+      "name": "Test Windows MSVC x86_64",
+      "filter": "windows-x64",
+      "extra_features": {
+        "wasmtime-wasi-nn": ["onnx"]
+      },
+    },
+    {
+      "os": "windows-latest",
+      "target": "x86_64-pc-windows-gnu",
+      "name": "Test Windows MinGW x86_64",
+      "filter": "mingw-x64"
+    },
+    {
+      "os": "ubuntu-latest",
+      "target": "aarch64-unknown-linux-gnu",
+      "gcc_package": "gcc-aarch64-linux-gnu",
+      "gcc": "aarch64-linux-gnu-gcc",
+      "qemu": "qemu-aarch64 -L /usr/aarch64-linux-gnu",
+      "qemu_target": "aarch64-linux-user",
+      "name": "Test Linux arm64",
+      "filter": "linux-arm64",
+      "isa": "aarch64",
+    },
+    {
+      "os": "ubuntu-latest",
+      "target": "s390x-unknown-linux-gnu",
+      "gcc_package": "gcc-s390x-linux-gnu",
+      "gcc": "s390x-linux-gnu-gcc",
+      "qemu": "qemu-s390x -L /usr/s390x-linux-gnu",
+      "qemu_target": "s390x-linux-user",
+      "name": "Test Linux s390x",
+      "filter": "linux-s390x",
+      "isa": "s390x"
+    },
+    {
+      "os": "ubuntu-latest",
+      "target": "riscv64gc-unknown-linux-gnu",
+      "gcc_package": "gcc-riscv64-linux-gnu",
+      "gcc": "riscv64-linux-gnu-gcc",
+      "qemu": "qemu-riscv64 -cpu rv64,v=true,vlen=256,vext_spec=v1.0,Zfa=true,zba=true,zbb=true,zbc=true,zbs=true,zbkb=true,zcb=true -L /usr/riscv64-linux-gnu",
+      "qemu_target": "riscv64-linux-user",
+      "name": "Test Linux riscv64",
+      "filter": "linux-riscv64",
+      "isa": "riscv64",
+    }
+  ];
+
+  for (let config of array) {
+    if (config.rust === undefined) {
+      config.rust = 'default';
+    }
+    if (config.extra_features === undefined) {
+      config.extra_features = {};
+    }
+  }
+
+  const filtered = array.filter(item => {
+    if (item.isa && names.includes(`cranelift/codegen/src/isa/${item.isa}`)) {
+      return true;
+    }
+    if (item.filter && commits.includes(`prtest:${item.filter}`)) {
+      return true;
+    }
+
+    // If any runtest was modified, re-run the whole test suite as those can
+    // target any backend.
+    if (names.includes(`cranelift/filetests/filetests/runtests`)) {
+      return true;
+    }
```

Comment:
This logic may not be quite so true any more given the more matrix entries we have, could the `item.isa` item be required here to basically only run the isa-based builders on runtest modifications? (e.g. not msrv)